src/main/webapp/app/controllers/submission/submissionListController.js

```diff
@@ -51,85 +53,142 @@ vireo.controller("SubmissionListController", function (NgTableParams, $controlle
         var start;
 
         var query = function () {
+            var pageNumber = $scope.page.number;
+            var pageCount = $scope.page.count;
+
+            if (angular.isDefined(sessionStorage.getItem("list-page-number")) && sessionStorage.getItem("list-page-number") !== null) {
+                pageNumber = sessionStorage.getItem("list-page-number");
+            }
+
+            if (angular.isDefined(sessionStorage.getItem("list-page-size")) && sessionStorage.getItem("list-page-size") !== null) {
+                pageCount = sessionStorage.getItem("list-page-size");
+            }
+
             $scope.tableParams = new NgTableParams({
-                page: $scope.page.number,
-                count: $scope.page.count
+                page: pageNumber,
+                count: pageCount
             }, {
                 counts: $scope.page.options,
                 total: $scope.page.totalElements,
                 filterDelay: 0,
                 getData: function (params) {
                     start = window.performance.now();
-                    return SubmissionRepo.query($scope.userColumns, params.page() > 0 ? params.page() - 1 : params.page(), params.count()).then(function (response) {
```

Comment:
What is the purpose of moving this?

Reply:
As described in the PR description:

> Despite the improvements, I still found a case where the page list is empty and the table is broken.
An example would be data changes since pagination is set.
It turns out that all of the data needed to determine if the selected pagination is correct is returned on the list response.
The problem is that the list response is already loaded so even if the pagination data were to be corrected there is no row data to display.
This is solved by making an extra request with the pagination data corrected using data from the first request.
Recursion is completely undesirable and so this extra request is performed exactly once and then only in the specific situation where the first request is incorrect.
Doing this required breaking apart the queryGetData() and adding a new parameter for forcing the page number.